src/main/java/mate/academy/bookstore/repository/book/BookSpecificationBuilder.java

```diff
@@ -0,0 +1,29 @@
+package mate.academy.bookstore.repository.book;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookstore.dto.BookSearchParametersDto;
+import mate.academy.bookstore.model.Book;
+import mate.academy.bookstore.repository.SpecificationBuilder;
+import mate.academy.bookstore.repository.SpecificationProviderManager;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class BookSpecificationBuilder implements SpecificationBuilder<Book> {
+    private final SpecificationProviderManager<Book> bookSpecificationProviderManager;
+
+    @Override
+    public Specification<Book> build(BookSearchParametersDto searchParameters) {
+        Specification<Book> spec = Specification.where(null);
+        if (searchParameters.authors() != null && searchParameters.authors().length > 0) {
+            spec = spec.and(bookSpecificationProviderManager.getSpecificationProvider("author")
+                    .getSpecification(searchParameters.authors()));
+        }
+        if (searchParameters.titles() != null && searchParameters.titles().length > 0) {
+            spec = spec.and(bookSpecificationProviderManager.getSpecificationProvider("title")
+                    .getSpecification(searchParameters.titles()));
+        }
```

Comment:
we can split into separate method this logical branches for better readability of the code